### PR TITLE
Polish code of circuit breaker module

### PR DESCRIPTION
**Pull request by sczyh30**
Signed-off-by: Eric Zhao <sczyh16@gmail.com>

### Describe what this PR does / why we need it

Polish code of circuit breaker module.

### Does this pull request fix one issue?

NONE

### Describe how you did it

Improve comments, naming and logging of circuit breaker module.

### Describe how to verify it

Run the test cases and demo.

### Special notes for reviews

NONE

### CODE REVIEW

File: core/circuitbreaker/circuit_breaker.go [hunk 1]
```diff
@@ -62,24 +64,29 @@ func (s *State) casState(expect State, update State) bool {
 }
 
 type StateChangeListener interface {
-	OnChangeToClosed(prev State, rule Rule)
+	// OnTransformToClosed is triggered when when circuit breaker state transformed to Closed.
```
Comment (louyuting) [inline core/circuitbreaker/circuit_breaker.go:67]: typo

File: core/circuitbreaker/circuit_breaker.go [hunk 2]
```diff
@@ -62,24 +64,29 @@ func (s *State) casState(expect State, update State) bool {
...
+	OnTransformToClosed(prev State, rule Rule)
 
-	OnChangeToOpen(prev State, rule Rule, snapshot interface{})
+	// OnTransformToOpen is triggered when when circuit breaker state transformed to Open.
```
Comment (louyuting) [inline core/circuitbreaker/circuit_breaker.go:70]: typo

File: core/circuitbreaker/circuit_breaker.go [hunk 3]
```diff
@@ -62,24 +64,29 @@ func (s *State) casState(expect State, update State) bool {
...
+	// The "snapshot" indicates the triggered value when the transformation occurs.
+	OnTransformToOpen(prev State, rule Rule, snapshot interface{})
 
-	OnChangeToHalfOpen(prev State, rule Rule)
+	// OnTransformToOpen is triggered when when circuit breaker state transformed to HalfOpen.
```
Comment (louyuting) [inline core/circuitbreaker/circuit_breaker.go:74]: typo

File: core/circuitbreaker/circuit_breaker_test.go [hunk 4]
```diff
@@ -29,7 +29,7 @@ func (m *CircuitBreakerMock) CurrentState() State {
 	return args.Get(0).(State)
 }
 
-// HandleCompleted handle the entry completed
+// OnRequestComplete handle the entry completed
 // rt: the response time this entry cost.
 func (m *CircuitBreakerMock) HandleCompleted(rt uint64, err error) {
```
Comment (louyuting) [inline core/circuitbreaker/circuit_breaker_test.go:34]: Also need to rename the HandleCompleted as OnRequestComplete in CircuitBreakerMock

File: core/circuitbreaker/circuit_breaker.go (hunk 1 above)
Comment (sczyh30) [reply core/circuitbreaker/circuit_breaker.go]: Fixed, please check.

File: core/circuitbreaker/circuit_breaker_test.go (hunk 4 above)
Comment (sczyh30) [reply core/circuitbreaker/circuit_breaker_test.go]: Updated, please check.

**Pull request merged**